Dockerfile

```diff
@@ -0,0 +1,39 @@
+ARG JAVA_VERSION=16
+ARG JVM_FLAVOR=hotspot
+
+FROM adoptopenjdk:${JAVA_VERSION}-jdk-${JVM_FLAVOR} AS builder
+WORKDIR /builddir
+
+COPY gradlew ./
+COPY gradle/ gradle/
+RUN ./gradlew --help --no-daemon # Download gradle and init
+
+COPY license*.txt build.gradle settings.gradle ./
+COPY src/ src/
+RUN ./gradlew clean buildForDocker --no-daemon
+
+
+ARG JAVA_VERSION
+ARG JVM_FLAVOR
+
+FROM adoptopenjdk:${JAVA_VERSION}-jre-${JVM_FLAVOR}
+WORKDIR /app
+
+RUN groupadd --system bibliothek \
+    && useradd --system bibliothek --gid bibliothek \
+    && chown -R bibliothek:bibliothek /app
+USER spring:spring
+
+VOLUME /data/storage
+EXPOSE 8080
+
+# We override default config location search path,
+# so that a custom file with defaults can be used
+# Normally would use environment variables,
+# but they take precedence over config file
+# https://docs.spring.io/spring-boot/docs/1.5.6.RELEASE/reference/html/boot-features-external-config.html
+ENV SPRING_CONFIG_LOCATION="optional:classpath:/,optional:classpath:/config/,file:./dockerdefaults.yaml,optional:file:./,optional:file:./config/"
+COPY ./docker/dockerdefaults.yaml ./dockerdefaults.yaml
```

Comment:
I would rather the file name be something like `default.application.yaml`.

Reply:
Sure. Wanted to avoid doing something like `application-default.yaml` because it seems like that's the file Spring normally looks for given no profile is enabled, but this should be fine.